src/Image.cc

```diff
@@ -75,6 +88,44 @@ struct WindowsBitmapHeader {
 } __attribute__((packed));
 
 
+
+static size_t init_bmp_header(WindowsBitmapHeader& header,
+    ssize_t width, ssize_t height, bool has_alpha,
+    size_t pixel_bytes, size_t row_padding_bytes) {
+  size_t header_size = sizeof(WindowsBitmapFileHeader) + (has_alpha ? sizeof(WindowsBitmapInfoHeader) : WindowsBitmapInfoHeader::SIZE24);
+
+  // Clear header so we don't have initialize fields that are 0
+  header = {};
+  header.file_header.magic = 0x4D42; // 'BM'
+  header.file_header.file_size = header_size + (width * height * pixel_bytes) + (row_padding_bytes * height);
+  header.file_header.reserved[0] = 0;
+  header.file_header.reserved[1] = 0;
+  header.file_header.data_offset = header_size;
+  header.info_header.header_size = header_size - sizeof(WindowsBitmapFileHeader);
+  header.info_header.width = width;
+  header.info_header.height = height;
+  header.info_header.num_planes = 1;
+  header.info_header.bit_depth = has_alpha ? 32 : 24;
+  header.info_header.compression = has_alpha ? 3 : 0; // BI_BITFIELDS or BI_RGB
+  header.info_header.image_size = has_alpha ? (width * height * 4) : 0; // 0 acceptable for BI_RGB
+  header.info_header.x_pixels_per_meter = 0x00000B12;
+  header.info_header.y_pixels_per_meter = 0x00000B12;
+  header.info_header.num_used_colors = 0;
+  header.info_header.num_important_colors = 0;
+  if (has_alpha) {
+    // Write V5 header
+    header.info_header.bitmask_r = 0x000000FF;
+    header.info_header.bitmask_g = 0x0000FF00;
+    header.info_header.bitmask_b = 0x00FF0000;
+    header.info_header.bitmask_a = 0xFF000000;
+    header.info_header.colorSpaceType = 0x73524742; // LCS_sRGB / 'sRGB'
+    header.info_header.renderIntent = 8;            // LCS_GM_ABS_COLORIMETRIC
```

Comment:
It looks like the fields after renderIntent aren't initialized anywhere - if they're actually unused, we should clear them here to avoid writing uninitialized memory to the file.

Reply:
For convenience, the complete header is set to 0 at the beginning:
```
header = {};
```